### PR TITLE
feat: convert records/config to TS

### PR DESCRIPTION
**Pull request by RobinCsl**
### fix: relax defaultVariant to a string

The reason is that there's nothing preventing some
default value being completely different than the
variant id's in the variant's config.

---

### feat: convert records/config to TS

NOTE: some tests were fixed in the process, some
unexpected behaviours were because the objects
passed to the function were not formed correctly.

This removed a few TODOs.

I still left a few of them with `@ts-expect-error`
to test for improper arguments. Not sure of their
value, but because we rely on Ramda so much, it's
probably a good thing to know what happens in these
edge cases (especially when we use `@types/ramda`
which could potentially be out of sync/incomplete
with the version of Ramda we're using).

### CODE REVIEW

File: src/records/config/index.test.ts [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+  ...STANDARD_VARIANTS,
+};
+
+describe("mapVariants", () => {
+  test("should convert variants to a standard form", () => {
+    expect(mapVariants(STANDARD_VARIANTS)).toStrictEqual(STANDARD_VARIANTS);
+    expect(mapVariants(NUMBER_VARIANTS)).toStrictEqual(NUMBER_VARIANTS_STANDARDISED);
+    expect(mapVariants(VARIANTS)).toStrictEqual(VARIANTS_STANDARDISED);
+  });
+  test("should return an empty object for unexpected inputs", () => {
```
Comment (jaroslav-kubicek) [inline src/records/config/index.test.ts:48]: not sure if I like that it does some magic instead of throwing error, but that's for another  discussion

**Review by jaroslav-kubicek — APPROVED** [review]

File: src/records/config/index.test.ts (hunk 1 above)
Comment (RobinCsl) [reply src/records/config/index.test.ts:48]: Yes, there's a lot of magic that I'd like to get rid of in future PRs.
